RHDEVDOCS-3306: Document running image build tasks as unprivileged builds (#44733)
openshift/openshift-docs: modules/op-security-with-custom-scc.adoc

[id='op-security-with-custom-scc_{context}']

> = Security with a custom SCC

Reviewer: ```suggestion
= Using a custom SCC to improve pod security
```
- This looks like it might be a procedure.

> You can improve security by using a custom <spell out what SCC stands for> (SCC) because it's easy to migrate pipelines that need to be run or built as root. To further lock down your installation, allow the container to run only as user 1000 when the pod asks for it and the default root is not allowed.

Reviewer: Please replace angle-bracketed content with real content.

> You can now leverage more security with custom SCC as it's easy to migrate pipelines that need to be run or built as root. However, if you would like to lock your installation further, you can only allow running the container as the user 1000 if the pod is asking for it and the default root is not allowed.

Reviewer: ```suggestion
You can improve security by using a custom <spell out what SCC stands for> (SCC) because it's easy to migrate pipelines that need to be run or built as root. To further lock down your installation, allow the container to run only as user 1000 when the pod asks for it and the default root is not allowed.
```

> 

Reviewer: ```suggestion
. Procedure
```
- Write the following instructions as numbered steps. 
- The introduction implies that there are multiple steps, but the following topic shows only one step.
- Or, if there is only one step, start the line with a bullet point, `* `, rather than a number, `. `.

> You can now leverage more security with custom SCC as it's easy to to migrate pipelines that needs to be run or build as root. However, if you like to further lock down your installation, you can only allow running the container as the user 1000 if the pod is asking for it and and the default root is not allowed.

Reviewer 2: to to

> You can now leverage more security with custom SCC as it's easy to to migrate pipelines that needs to be run or build as root. However, if you like to further lock down your installation, you can only allow running the container as the user 1000 if the pod is asking for it and and the default root is not allowed.

Reviewer 2: to to

To do so you need to edit the `pipelines-scc` and modify the `runAsUser` and `seLinuxContext` section to `MustRunAs` to uid 1000.

[source,yaml,subs="attributes+"]
----
runAsUser:
  type: MustRunAs
  uid: 1000
seLinuxContext:
  type: MustRunAs
----

> The pipeline `serviceaccount` will only allow running images as user 1000 and not as root anymore.

Reviewer: The word "will" is a red flag. Whenever possible, avoid using the future tense in technical writing. Rather, use the present tense.
```suggestion
The pipeline `serviceaccount` allows images to run only as user 1000, not as root.
```
